packages/realm-server/tests/server-endpoints-test.ts

```diff
@@ -1501,6 +1504,340 @@ module(basename(__filename), function () {
+
+          // Republish
+          let secondResponse = await request2
+            .post('/_publish-realm')
+            .set('Accept', 'application/vnd.api+json')
+            .set('Content-Type', 'application/json')
+            .set(
+              'Authorization',
+              `Bearer ${createRealmServerJWT(
+                { user: ownerUserId, sessionRoom: 'session-room-test' },
+                realmSecretSeed,
+              )}`,
+            )
+            .send(
+              JSON.stringify({
+                sourceRealmURL: testRealm2.url,
+                publishedRealmURL: 'http://testuser.localhost/test-realm/',
+              }),
+            );
+
+          assert.strictEqual(secondResponse.status, 201, 'Republish succeeds');
+          assert.strictEqual(
+            secondResponse.body.data.id,
+            firstResponse.body.data.id,
+            'Same published realm ID',
+          );
+          assert.strictEqual(
+            secondResponse.body.data.attributes.publishedRealmURL,
+            firstResponse.body.data.attributes.publishedRealmURL,
+            'Same published realm URL',
+          );
+          assert.notEqual(
+            secondResponse.body.data.attributes.lastPublishedAt,
+            firstTimestamp,
+            'Timestamp is updated on republish',
+          );
+        });
+
+        test('POST /_publish-realm returns bad request for missing sourceRealmURL', async function (assert) {
+          let response = await request2
+            .post('/_publish-realm')
+            .set('Accept', 'application/vnd.api+json')
+            .set('Content-Type', 'application/json')
+            .set(
+              'Authorization',
+              `Bearer ${createRealmServerJWT(
+                { user: ownerUserId, sessionRoom: 'session-room-test' },
+                realmSecretSeed,
+              )}`,
+            )
+            .send(JSON.stringify({}));
+
+          assert.strictEqual(response.status, 400, 'HTTP 400 status');
+          assert.strictEqual(
+            response.text,
+            '{"errors":["sourceRealmURL is required"]}',
+            'Error message is correct',
+          );
+        });
+
+        test('POST /_publish-realm returns bad request for missing publishedRealmURL', async function (assert) {
+          let response = await request2
+            .post('/_publish-realm')
+            .set('Accept', 'application/vnd.api+json')
+            .set('Content-Type', 'application/json')
+            .set(
+              'Authorization',
+              `Bearer ${createRealmServerJWT(
+                { user: ownerUserId, sessionRoom: 'session-room-test' },
+                realmSecretSeed,
+              )}`,
+            )
+            .send(
+              JSON.stringify({
+                sourceRealmURL: testRealm2.url,
+              }),
+            );
+
+          assert.strictEqual(response.status, 400, 'HTTP 400 status');
```

Comment:
I’m deploying and will review more thoroughly after that but I think this should be 422, 400 is more like bad syntax, where this is correct JSON but an invalid value.